### DISCUSSION TITLE
Werte OpenDTU und Ahoy DTU unterschiedlich?!?

### DISCUSSION THREAD

**Discussion by 3003M** [hoylabs/OpenDTU-OnBattery#1011] [Q&A]
Servus,

ich bin ganz neu bei OpenDTU, seit vorgestern läuft testweise Opendtu Onbattey.
Ich habe schon seit längerem eine Ahoy am laufen und einen Shelly hinter dem Hoymiles.
Mir ist jetzt aufgefallen dass OpenDTU andere Yield Werte anzeigt als Ahoy und Shelly.
Beim Ahoy und Shelly ist der Gesamtertrag ca gleich bei ~960KWH, OpenDTU zeigt aber 1009KWH.

Bei Ahoy gibt es ja die Funktion "Yield Efficiency" die irgendwo bei 95% liegt, wenn ich diese auf 100% stelle habe ich auch die 1009KWH.
Gibt es diese Einstellung bei OpenDTU Onbattery nicht oder bin ich zu blöd die zu finden? Es wird ja in der Live Ansicht der Wirkungsgrad angezeigt der auch ~95% hat, das wird aber nicht auf den Gesamt-/ Tagesertrag gerechnet?
Ich denke dass die 960KWH eher stimmen, da der Shelly das ja auch anzeigt.

Wäre für jede Idee dankbar. :)

**Comment by spcqike** [comment]
Du sagst also, dass der vom Wechselrichter gemeldete Wert nicht effizienzbereinigt ist?

Der Tagesertrag, der Gesamtertrag, die Effizienz, ... das sind alles Werte, die der Wechelrichter selbst ausgibt. Bisher gehe ich davon aus, dass diese, im Rahmen allgemeiner Messtoleranzen, passen.

Ich hab mal meine Statistik von gestern grob überflogen und kann dort keine Auffälligkeit feststellen.

Mein HMS-2000 hat in 6h 5.97kWh gemeldet. 

im gleichen Zeitraum war der Verbrauch auf der Phase bei ~15W (so genau lässt sich das im Laufe des Tages nicht ablesen, nachts, also ohne PV, sind es aber sehr konstante 15W, ein Kühlschrank oder Ähnliches ist dort nicht angeklemmt), also 0.09kWh. 
Einspeisend gemeldet hat der Shelly 3EM 5.88kWh.

Fast schon gruselig und irgendwie .... unglaubwürdig ..... lässt es mich doch selbst an meinen Daten zweifeln (hab ich was übersehen?), aber es passt ganz genau zusammen. 5.88 Einspeisung + 0.09 direkter Eigenverbrauch = 5.97 Erzeugung durch den Hoymiles.


Warum sollte der Hoymiles den Tagesertrag eigentlich auch nicht korrekt berechnen? Immerhin ist die, von ihm selbst ermittelte Effizienz, ja das Ergebnis aus DC-Eingangsleistung und AC-Ausgangsleistung. Das setzt doch bereits voraus, dass der AC-Ausgang annähernd korrekte Werte liefert.

**Comment by 3003M** [comment]
Ich dachte auch dass das alles Werte sind die vom Wechselrichter kommen, aber es wird bei den beiden unterschiedlich angezeigt.
Scheinbar sind die angezeigten Werte tatsächlich nicht effizienzbereinigt?!?


![20240530_095430](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/143808310/e2ad0e18-e113-4660-a6e0-33c7e4ead7dc)
Links OpenDTU, rechts Ahoy.

Und nein, die sind nicht immer parallel an, ich habe die Ahoy nur für das Bild eingeschaltet.

Edit: Shelly zeigt übrigens 955KWH.

Offtopic: Was ist das eigentlich für ein Streifen links im Display bei OpenDTU?

**Reply by spcqike** [reply]
> Offtopic: Was ist das eigentlich für ein Streifen links im Display bei OpenDTU?

ich meine, das passiert, wenn man den falschen OLED Treiber auswählt. Ich hatte sowas auch irgendwann mal.
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/143165e8-0ddf-432f-a7d8-3b58e7e7dbfb)


Ich kann die Abweichungen bei meinem bisherigen Aufbau nicht nachvollziehen. Für mich ergibt es auch wenig Sinn.

Haben andere User auch die Möglichkeit, die Erzeugung an zwei unterschiedlichen Messwertgebern zu vergleichen?

Ich könnte theoretisch noch ne smarte Steckdose nehmen, oder einen weiteren Shelly in die Leitung hängen. Besser machts das aber auch alles nicht. Immerhin erhöht jedes Gerät den Widerstand und erzeugt eigene Verluste.

bei voller Einspeisung hab ich ~30W Differenz zwischen Hoymiles Erzeugung und Shelly 3EM Einspeisung.
2040W zu 2010W. Nehme ich noch meine 15W dauerhaften Standbyverbrauch auf der Phase an, liegen sie gar nur noch 15W auseinander. (von den 2040W würden ja 15W direkt verbraucht, also können nur noch 2025W eingespeist werden)

Auch ohne die 15W Eigenverbrauch, angenommen 30W effektive Differenz zwischen beiden Zählern. Dazwischen liegt ja noch einiges an Leitung/Stecker/Verbindungen. In meinem Fall ~ 30m Kabel (mehrere, von der HV mit Shelly erst in die Werkstatt, dann weiter in die Garage), mehrere LS, ein RCD, diverse Verbindungen ..... 
bei 30W und 8.5A (2040W am Hoymiles bei 240V) macht das, frei nach P=UI 3.5V Spannungsabfall bzw. nach U=RI 0.4Ohm Gesamtwiderstand.

gem. VDE

> Grenzwerte der Norm VDE 0701-0702 
Für Geräte bis 16 A Nennstrom (d. h. üblicherweise Schutzkontakt-Stecker oder „Drehstrom“-Anschluss CEE 16 A) gilt: maximal 0,3 Ω für Anschlussleitungen bis 5 m und. bei längeren Leitungen kann für jede weitere 7,5 m 0,1 Ω dazu gerechnet werden.

bin ich da also gut dabei.

und:
würde ich die AC-Leistung des Hoymiles um die gemeldete Effizienz nach unten korrigieren, würde ich am Shelly 3EM mehr einspeisen, als der Hoymiles überhaupt produziert 😄 


Wie auch immer, wer viel misst, misst Mist! :) für mich haben die Daten vom Hoymiles bisher anstandslos gepasst. (ich glaub auch, dass der DPL sonst mehr Probleme machen würde, oder?)


Nachtrag:
Dabei fällt mir ein. Wo misst du denn mit dem Shelly? direkt an der Steckdose wo du auch einspeist, oder in der zentralen Verteilung? Wenn in der Verteilung, wie weit ist sie weg? Sind die Differenzen vielleicht einfach Verluste, die auf der Leitung bis zum Shelly auftreten? meine überschlagenen 0.4Ohm Leitungswiderstand sind nach VDE, so sie denn stimmen, ja eher ideal und sehr gering, bezogen auf die Leitungslänge. bei 30m Anschlussleitung dürfte ich ja 0.6-0.7Ohm haben. Bestandsbauten und alten Leitungen haben ggf. weit mehr. Who knows?

**Reply by 3003M** [reply]
Das mit dem Treiber für das Display werde ich testen, vielen Dank!

Der Shelly sitzt bei mir ca 5m hiter dem Wechselrichter (2,5mm²) in einer Verteilerdose, an der Leitung hängt sonst nichts dran, die geht direkt in den Zählerschrank.
Das bischen Abweichung zw. Shelly und Ahoy ist auch vernachlässigbar, mich hat halt nur verwundert dass die OpenDTU im Vergleich zur Ahoy so andere Werte zeigt, bzw die Effizient scheinbar nicht mit einrechnet. Wenn man die ~5% Verlust mit einrechnet stimmen die Werte ja überein, nur da Ahoy näher am Ertrag vom Shelly ist vermute ich halt dass die Daten vom Ahoy eher stimmen als von der OpenDTU.

Zur Info, benutze die Firmwareversion 2024.05.07

**Comment by 3003M** [comment]
Hat jemand sonst evtl noch eine Idee warum mir zu hohe Werte angezeigt werden?
Sollte ich mal eine ältere FW Version testen?

**Reply by spcqike** [reply]
> Sollte ich mal eine ältere FW Version testen?

kannst du probieren. Würde mich aber wundern wenn es was ändert da openDTU, soweit ich weiß, bisher immer die gemeldeten Werte vom wechselrichter 1:1 weiter geleitet hat. 

Wie gesagt, alles andere macht halt auch wenig Sinn. 

Mein wechselrichter meldet grade 214W_AC von 225W_DC und eine Effizienz von ~95% 

verrechne ich den AC wert nun erneut mit der Effizienz, kämen nur noch 204W_AC raus. Das wären traurige 90% Effizienz. Und es wäre auch weniger als der Shelly 3EM auf L2 als Einspeisung meldet. 

Interessieren würde mich aber dennoch, was andere User für Erfahrungen gemacht haben :)

**Reply by Solarteur** [reply]
Die Momentanwerte der **Leistung** sind (im Rahmen der Messgenauigkeit) richtig und auch als AC bzw. DC gekennzeichnet. Auch die Übereinstimmung mit der Anzeige eines Shelly EM ist gut

"Problem" sind sind die **Ertrags**werte (YieldTotal bzw. YieldDay).

In Ahoy gibt es (oder zumindest gab es) die Möglichkeit, im Menu einen Faktor einzugeben, z. B. 0,955 (=Durchschnittwirkungsgrad Hoymiles WR), um die auch aus meiner Sicht DC-Werte beim **Ertrag** näherungsweise in AC-Werte umzurechnen.  Ich habe z. Z. kein Ahoy-System verfügbar, weil ich auf OpenDTU.onBattery umgestiegen bin. Ich erinnere mich aber daran, dass mit einem Faktor 1 die Werte identisch waren (das war vor ca. 6 - 9 Monaten). Meine Messung des AC-Ertrages ('Einspeisung) mit einem Shelly EM liefert Werte, die so 2% unter den Werten für den kumulierten Ertrag von OpenDTU (DC?) liegen. 

Über dieses Thema gab es schon viele Diskussionen, z. B. https://github.com/lumapu/ahoy/issues/580
Leider ist der Link zum Manual tot.

Gerade die Tatsache, dass die Summe aus den blauen DC-Feldern sich im grünen AC-Feld als "YieldTotal" bzw. YieldDay" wiederfindet, spricht aus meiner Sicht für DC-Werte, auch wenn es nicht unbedingt Sinn macht. Die Tabelle von stefan123t ist eigentlich eindeutig.

In anderen Beiträgen - leider finde ich den Link mehr - wurde die Momentanleistung durch externe Messgeräte über 24h integriert. Diese Messungen sprachen eher dafür, dass bei Ahoy und OpenDTU bei den Ertragswerten AC-Werte ausgeben werden. 

Ich verwende OpenDTU-onBattery zum geregelten Laden-/Entladen eines Akkus und mache die "Bilanzierung" über die Daten eines Shelly EM.

**Reply by spcqike** [reply]
ok. das erschließt sich mehr eher. Ich kann nachvollziehen, dass der Tagesertrag die Summe der Tageserträge der Eingänge ist.

allerdings kann ich nicht nachvollziehen, dass der Tagesertrag vom extern gemessenen Ertrag abweicht.

bspw. am heutigen Tag:
mein WR meldet von 07:15 bis 14:15 eine Erzeugung von 2.88kWh. er speist auf L2 ein.
auf L2 habe ich in der Zeit nur meinen Standbyverbrauch von ~10W. (ich muss meine Aussage aus https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/1011#discussioncomment-9604044 revidieren. mein Verbrauch nachts liegt bei 15W, da dort aber auch das beleuchtete Hausnummernschild dran hängt, ist es tagsüber natürlich weniger. am Tag sind es ~11W)
in den 7 h hab ich also ~0.078kWh direkt verbraucht.
der Shelly 3EM misst einspeisend auf L2 noch 2.80kWh. 

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/443589e6-c9b8-43f5-a9a2-9dd3e88607cf)
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/52fe475a-30ca-4c90-8976-40c8ca2b270f)


auch die Mittelwerte der Graphen (-401W und +412W) zeigen den direkten Eigenverbrauch von ~10-11W.
Den Leistungssprung sieht man bspw. abends, wenn die PV bereits nichts mehr liefert und die Beleuchtung an geht
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/bd85e13f-81ed-4539-9ffe-760c543a2a3d)


zumindest in meiner Aufzeichnung vom Hoymiles und vom Shelly 3EM passen sowohl die Leistungen als auch die Erträge sehr gut zusammen. 

Sind die Tageserträge vielleicht bereits im Hoymiles effizienzbereinigt?

Ich glaube, ich werde in den kommenden Tagen doch noch mal ein Schätzeisen dazwischen bauen. Sei es eine Tasmota Steckdose, ein Shelly PM oder mein ELV Energy Meter.... oder alle .... aber was mache ich, wenn sie sich nicht einig werden 😄

**Reply by Solarteur** [reply]
Eine allgemeine Anmerkung: Wir sollten bei allem "Streben nach der Genauigkeit" nicht vergessen, wie hoch die Messfehler bei "offiziellen" Stromzählern sein dürfen - einfach um unsere Diskussion über die Messung der kWh in ein realistisches Umfeld zu stellen.

Laut https://de.wikipedia.org/wiki/Stromz%C3%A4hler sind die Eichfehlergrenzen

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/117977338/1d5b57af-52e5-488a-b165-e3439bf6529a)

Die erlaubten [Verkehrsfehlergrenzen](https://de.wikipedia.org/wiki/Verkehrsfehlergrenze) des Zählers (die während der Betriebsdauer des Zählers auftreten dürfen) sind doppelt so hoch, wie die oben angegebenen [Eichfehlergrenzen](https://de.wikipedia.org/wiki/Eichfehlergrenze).

Bei mir ist ein Zähler der Genauigkeitsklasse A eingebaut...

@3003M: Ich habe den **Eingangspost** jetzt genau durchgelesen:
- Die "Yield Efficiency" von Ahoy gibt es bei OpenDTU/OpenDTU-onBattery nicht,
- gut zu wissen, dass Ahoy ohne Korrekturfaktor und OpenDTU/OpenDTU-onBattery letztlich identische Werte aus den Wechslrichtern auslesen,
- alles in allen passt das in das Bild, dass die kumulierten Erträge DC-Werte sind, bei Ahoy über die Yield Efficiency "in Richtung AC-Werte" korrigiert werden können, bei OpenDTU/OpenDTU-onBattery aber nicht.

**Comment by 3003M** [comment]
Hi

sorry für die späte Antwort, konnte nicht früher schreiben.

Vielen Dank für eure ausführlichen Antworten! 
Ich wollte da auch keine Wissenschaft draus machen, ich war nur verwundert nachdem ich ein Jahr Ahoy benutzt habe dann bei OpenDTU ganz andere Werte zu sehen.
Für die (bei mir noch kommende) Steuerung wird das nichts ausmachen, ich schaue aber gerne auf das Display und da immer 5% abziehen zu müssen ist halt blöd. Ist aber so, kann man nix machen, ist ja auch kein Weltuntergang.

@Solarteur Wow, das ist ja heftig, bis zu 8% Ungenauigkeit bei einphasiger Last, und das ist leider mein Szenario. Hier ist alles, warum auch immer, auf nur einer Phase von 3 angeschlossen... Da hoffe ich doch dass die max 8% zu meinen Gunsten gehen ;D

Nochmals vielen Dank für eure Mühe und falls @spcqike seine Messungen wirklich macht freue ich mich hier auf die Ergebnisse! :)

**Reply by spcqike** [reply]
Ich hatte das schon wieder ganz vergessen, da die Werte bei mir scheinbar passen :) 

ja, Toleranzen und Unsicherheiten hat man sicher überall. Wie immer „wer viel misst misst Mist“ 

ich hab jetzt den wechselrichter, den ich in der openDTU auslesen kann, den Shelly 3EM in der Verteilung am Zähler, eine Gosund SP1 geflasht mit tasmota und kalibriert mit einem kalibrierten Multimeter und den ELV Strommesser. 

Lange werd ich das so nicht betreiben, 3 Geräte nacheinander in der Steckdose ist irgendwie blöd :D (ELV-Gosund-Hoymiles) aber für n test wird es schon gehen. 

In den ersten 10 Minuten, die der WR nun gebraucht hat im wieder seine maximal Leistung zu erreichen, passt alles 1:1 zusammen. Die DTU sagt 668W (ne Wolke) und 291Wh, das gleiche sagt auch die Gosund SP1. 

Den ELV werd ich heute Abend vielleicht auslesen. Dafür muss ich ja vor Ort gehen :)

**Reply by spcqike** [reply]
Hier ein paar Bilder 

die Gosund scheint sogar mehr zu zählen als der wechselrichter 

![IMG_6082](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/246c11eb-799f-4f83-b596-9c166244683b)
![IMG_6083](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/fb89f473-16b9-4206-af95-fe23f8fd8aa2)
![IMG_6084](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/1d538c9d-8128-407a-b2d4-bb9e5c04e254)

**Reply by spcqike** [reply]
Letzter Nachtrag bis heut Abend:

der ELV misst keine Energie wenn eingespeist wird der Zähler ist auf 0.000. allerdings sind sich der ELV und die Gosund einig, was Leistung, Scheinleistung, Strom und so weiter betrifft. 

Der wechselrichter auf der anderen Seite meldet mehr Leistung. 

Im Raum stehen bei voller Leistung 2040W (hoymiles) gegenüber 2022W (Gosund und ELV) gegenüber 2000W (Shelly 3EM)

so, mal gucken was die Systeme heute Abend zum besten geben. :)

**Reply by spcqike** [reply]
So. Abend Update. 

Seit Installation und entsprechenden Neustart: hoymiles meldet 9.02kWh bei aktuell noch 158W. Die Gosund meldet 8.97kWh bei aktuellen 156W. ELV meldet ebenfalls 156W aber leider keine Energie. 

50Wh Differenz, also ~0.5% Abweichung, sind für mich nicht signifikant. Zumal der Gosund aufgrund der zusätzlichen Leitung und klemmstelle weniger messen muss. 

Für mich passt es also, ich vertraue den Werten weiterhin, ohne sie zu korrigieren. 

Das kann natürlich durch fertigungstoleranzen und erlaubten Toleranzen / Abweichungen aller Messsysteme bei anderen Benutzern anders aussehen. 

![IMG_6089](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/5cbf4573-0976-4916-b45c-b2712166b2fe)
![IMG_6086](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/deedd7c4-04c9-4759-838f-850ccf8cac2d)
![IMG_6087](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/e9fe2605-400a-45bc-85e2-203a38fa72b8)

**Comment by spcqike** [comment]
@schlimmchen @Manos1966 habt ihr sowas bei euch auch beobachtet? 

Ich find den Hintergrund eigentlich interessant und überlege, ob es nicht doch sinnvoll ist, einen Faktor einzubauen. Bei mir scheint die Energie ganz gut hinzukommen. zumindest sehe ich bei mir grad keinen Bedarf, die gemeldete Energie zu korrigieren. Bei anderen Baureihen / Chargen / Toleranzen der Komponenten kann das aber ganz anders aussehen.

wobei, wenn man sowas angeht, gehört der Faktor wahrscheinlich in die Config eines jeden Wechselrichters, und damit eher in den Upstream, als in die openDTU-onBattery, oder?

**Reply by schlimmchen** [reply]
> habt ihr sowas bei euch auch beobachtet?

Noch nicht geschaut. Ich hab einen "Erzeugungszähler" eingebaut, aber mit Tasmota und selbst kalibriert. Wenn es da Abweichungen gibt weiß ich nicht, woher sie kommen...

Das gehört dann in den Upstream. Aber eigentlich gehört's in den WR.

**Reply by Manos1966** [reply]
Hi @spcqike 
ich habe mit Shelly 1PM, Tasmota, Shelly 3em immer leicht unterschiedliche Ergebnisse beobachtet.
Der Unterschied zwischen AHOY, OpenDTU und SHELLY ist gravierend, je hoeher die Produktion ist (siehe Dach  46W 5% Abweichung, waehrend Zaun 12W 3% Abweichung)
![Unterschied Shelly OpenDTU](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/fe53d362-fe25-4b32-b926-26b2191b7568)

Am Ende habe ich aufgegeben. Die einzige Zahl die zaehlt, ist die des digitalen Sromzaehlers (und sie muss negativ bleiben) 😉 

Nebeninfo: Die original DTU von Hoymiles funktioniert so, dass nur jede 15 Min die Daten aktualisiert werden.
Daher bin ich heute sehr zufrieden, dass die Hoymiles sich so schnell heute regeln laesst... in ein paar Sekunden!
Wir "spielen" mit einem Geraet, das nie fuer eine solche Aufgabe entwickelt wurde.
